util/uri/src/uri.rs

```diff
@@ -11,8 +11,8 @@ use url::Url;
 
 #[derive(Debug, Display)]
 pub enum UriParseError {
-    /// Url parse error: {0}
-    UrlParse(url::ParseError),
+    /// Url parse error: {0}, "{1}"
```

Comment:
I think it makes more sense to have the url first and the error second and then have the message me `Error passing URL {0}: {1}`. This applies to other places below as well.